examples/abbas_work_area/check_thread.py

```diff
@@ -0,0 +1,93 @@
+import time
+import threading
+import glob
+import os
+import hashlib
+import json
+
+
+directory = "./multithreading/*"
+content_lock = threading.Lock()
+content = []
+data_store = "./data/data.json"
+
+
+def process_file(file_path, data="", state="r"):
+    with open(file_path, state) as store_file:
+        if state == "a":
+            store_file.write(data)
+        elif state == "r":
+            return store_file.read()
+
+
+def get_existing_hashes():
+    existing_hashes = set()
+    file_content = process_file(data_store)
+    if file_content:
+        for line in file_content.splitlines():
+            store_entry = json.loads(line)
+            existing_hashes.add(store_entry['hash'])
+        return existing_hashes
+
+
+def is_empty(path):
+    files = glob.glob(path)
+    return len(files) == 0
+
+
+def check_dir():
+    while True:
+        if not is_empty(directory):
+            check_if_read()
+        time.sleep(2)
+
+
+def check_if_read():
+    files = glob.glob(directory)
+    existing_hashes = get_existing_hashes() or set()
```

Comment:
If we load hashes from a file while processing each and every file, it can add up in a lot of delay with increasing number of files.